paimon-flink/paimon-flink-common/src/test/java/org/apache/paimon/flink/BatchFileStoreITCase.java

```diff
@@ -84,6 +84,25 @@ public void testOverwriteEmpty() {
         assertThat(batchSql("SELECT * FROM T")).isEmpty();
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testOverwritePartitionedTableEmpty(boolean dynamicPartitionOverwrite) {
+        batchSql("CREATE TABLE T2 (a INT, b INT, c INT) PARTITIONED BY (a) ");
+        batchSql("INSERT INTO T2 VALUES (1, 11, 111), (2, 22, 222)");
+        assertThat(batchSql("SELECT * FROM T2"))
+                .containsExactlyInAnyOrder(Row.of(1, 11, 111), Row.of(2, 22, 222));
+        batchSql(
+                "INSERT OVERWRITE T2 /*+ OPTIONS('dynamic-partition-overwrite' = '%s') */"
```

Comment:
I don't think this UT covers the situation that appendTableFiles contains some DELETE files.

Reply:
It will cover. But, I take an another thought, I think the current behavior is acceptable, It's consistent for the partitioned and unpartitioned table when insert overwrite with false. So, closing it now.